### PR TITLE
Fix specs and build; invocations changed in Pry and Rspec3

### PR DESCRIPTION
**Pull request by joallard**
@ConradIrwin The only thing is, do you have a better idea for the `run_command` invocation? You surely know the Pry test helper usage better than me, so you might have something interesting. Otherwise, pretty much easy fixes.

### CODE REVIEW

**Comment by joallard** [comment]
Yikes, Capybara 2.8 requires Rack 2.0 requires Ruby >= 2.2.

**Comment by ConradIrwin** [comment]
Thanks!. Ook...

Going to merge these as is.

**Comment by ConradIrwin** [comment]
p.s. I've made you a collaborator — please feel free to send PRs for big changes, but I trust you to tweak small things :).

Thanks so much!

**Comment by joallard** [comment]
Will do, I'm not even game to send big PRs to my own repos! 😆 I'm quite honored, I'm a big Pry and Rescue fan! (That + rspec + a vim hotkey to rescue = feels awesome)

@ConradIrwin about the Rack dependency, should we deprecate some Rubies, or possibly test against several Gemfiles? You have a deprecation guideline in mind?

**Comment by ConradIrwin** [comment]
I haven't been using ruby for the last year or so, so not sure how much the landscape has changed. It's almost certainly find to drop 1.8, and probably 1.9.Is anyone stuck on 2.1, or has everyone upgraded to 2.2/2.3 already?ConradSent via SuperhumanOn Thu, Sep 15, 2016 at 3:00 PM, Jonathan Allard<notifications@github.com>wrote:Will do, I'm not even game to send big PRs to my own repos! 😆 I'm quite honored, I'm a big Pry and Rescue fan! (That + rspec + a vim hotkey to rescue = feels awesome)

@ConradIrwin about the Rack dependency, should we deprecate some Rubies, or possibly test against several Gemfiles? You have a deprecation guideline in mind?—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by joallard** [comment]
For starters, the new Rails 5 depends on Ruby 2.2.2+.

Further, 2.0 is end-of-life, Ruby-core won't even release security fixes for it anymore.

> As it has been announced before, all support for Ruby 2.0.0 has ended today \[24 Feb 2016]. Bug and security fixes from more recent Ruby versions will no longer be backported to 2.0.0, and no further patch release of 2.0.0 will be released.

https://www.ruby-lang.org/en/news/2016/02/24/support-plan-of-ruby-2-0-0-and-2-1/

And 2.1 has stopped getting bugfixes, with the release of 2.1.10.

> As announced before, this is the last normal release of the Ruby 2.1 series. After this release we will never backport any bug fixes to 2.1 except security fixes. We recommend that you start planning to upgrade to Ruby 2.3 or 2.2.

https://www.ruby-lang.org/en/news/2016/03/30/ruby-2-1-9-released/

Hunted down the Ruby maintenance policy, with the following:

> MINOR: With "best effort" goal with at least 6 months and at best 3 years or after two newer MINOR releases.

> 
> [...]
> 
> Once a version is End-of-Life, it's your responsibility to deal with bugs and security fixes. An approved commercial party may be able to accept responsibility of maintaining an End-of-Life version.
> 
> It's important to recognize the maintainer for each version may decide they can no longer support a version, in which case ruby-core would work to responsibly discontinue the version.

https://bugs.ruby-lang.org/projects/ruby/wiki/GeneralMaintenancePolicy

So, to sum up:

* 2.4: not yet released, at `2.4.0.preview2`
* 2.3: supported, 2.3.1
* 2.2: supported, 2.2.5
* 2.1: security-only maintenance, EOL soon, 2.1.10
* 2.0: not supported by core anymore.

@ConradIrwin seems that we could deprecate at least <=2.0 and add tests for 2.1–2.3, 2.4 allowed fail. Which would leave a version of Capyba-rack for 2.2+, and another for 2.1.

PS. Your email replies have wacky formatting on GH.

**Comment by ConradIrwin** [comment]
That sounds good to me.

Oops, the perils of building a new email client... I wonder what undocumented magic I need to do to make Github parse it properly :)

**Comment by joallard** [comment]
@ConradIrwin Would you kindly make another release? Heroku has been preventing me to push anytime I got 'master' in my Gemfile. :/
